src/coreclr/gc/gc.cpp

```diff
@@ -3729,7 +3738,62 @@ uint8_t* region_allocator::allocate (uint32_t num_units, allocate_direction dire
     if (alloc)
     {
         total_free_units -= num_units;
-        print_map ("alloc: found at the end");
+        if ((fn != nullptr) && (global_region_left_used > bookkeeping_data_committed))
```

Comment:
seems like this could totally be done outside the `region_allocator`, if you are already passing in a function pointer to do the commit, the point would be to not have `region_allocator` care about this commit at all. why not do this in the function you passed in? if it fails to commit just call `delete_region`.